dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/internal/domain/event/LegacyToSpanEventMapper.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.tracing.internal.domain.event
+
+import com.datadog.android.BuildConfig
+import com.datadog.android.core.internal.CoreFeature
+import com.datadog.android.core.internal.Mapper
+import com.datadog.android.core.internal.net.info.NetworkInfoProvider
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.core.internal.utils.toHexString
+import com.datadog.android.log.internal.user.UserInfoProvider
+import com.datadog.android.tracing.model.SpanEvent
+import com.datadog.opentracing.DDSpan
+
+internal class LegacyToSpanEventMapper(
+    private val timeProvider: TimeProvider,
+    private val networkInfoProvider: NetworkInfoProvider,
+    private val userInfoProvider: UserInfoProvider
+) : Mapper<DDSpan, SpanEvent> {
+
+    // region Mapper
+
+    override fun map(model: DDSpan): SpanEvent {
+        val serverOffset = timeProvider.getServerOffsetNanos()
+        val metrics = resolveMetrics(model)
+        val metadata = resolveMeta(model)
+        return SpanEvent(
+            traceId = model.traceId.toHexString(),
+            spanId = model.spanId.toHexString(),
+            parentId = model.parentId.toHexString(),
+            resource = model.resourceName,
+            name = model.operationName,
+            service = model.serviceName,
+            duration = model.durationNano,
+            start = model.startTime + serverOffset,
+            error = if (model.isError) 1 else 0,
+            meta = metadata,
+            metrics = metrics
+        )
+    }
+
+    // endregion
+
+    // region internal
+
+    private fun resolveMetrics(event: DDSpan) = SpanEvent.Metrics(
+        topLevel = if (event.parentId.toLong() == 0L) 1 else null,
+        additionalProperties = event.metrics
+    )
+
+    private fun resolveMeta(event: DDSpan): SpanEvent.Meta {
+        val networkInfo = networkInfoProvider.getLatestNetworkInfo()
+        val simCarrier = SpanEvent.SimCarrier(
+            id = networkInfo.carrierId.toStringOrNull(),
+            name = networkInfo.carrierName
+        )
+        val networkInfoClient = SpanEvent.Client(
+            simCarrier = simCarrier,
+            signalStrength = networkInfo.strength.toStringOrNull(),
```

Comment:
The way I understood from the code, no but will double check this.

Reply:
Actually yes, it can be negative -> https://app.datadoghq.com/logs?agg_m=count&agg_q=&agg_t=count&cols=%40org_details.org_name%2C%40org_id%2Cservice%2C%40network.client.signal_strength&from_ts=1617609571000&index=&live=true&messageDisplay=inline&query=source%3Aandroid&sort_m=&sort_t=&stream_sort=desc&to_ts=1617695971000&top_n=&top_o=&viz=stream

Reply:
yes, because normally signal strength for mobile networks is measured in dBm with values on the negative scale.

Reply:
Yup, it is fixed in the next commit.